### ISSUE TITLE
[220113]Image Segmentation Using Deep learning A survey [1]

### ISSUE THREAD

**Issue by ChaeChae0505** [ChaeChae0505/TIL_STUDY#6]
- 논문 : Image Segmentation Using Deep learning A survey
- 분류 : Deep learning review
- 저자 : Shervin Minaee, Yuri Boykov, Fatih Porikli, Antonio Plaza, Nasser Kehtarnavaz, and Demetri Terzopoulos
- 읽는 배경 :  정신 차리고 survey 부터 다시 읽어 보기로 했다. 그 중에 제일 모른다고 생각하는 survey 부터 읽었다
- 느낀점 : 리뷰논문은 나눠서 읽자 한번에 다 읽기 너무 지친다
- 공부 참고자료

---

# Abstract

→ 꼼꼼하게 볼 부분 밑줄

- Image segementation은 CV 와 Image processing에 있어서 적용되는 곳이 많다 . Scene understanding 의학 분야, 로봇 분야, 비디오, AR, image compression 등등 널리 적용 된다. 다양한 알고리즘이 있는데 최근에는 딥러닝을 활용해서 성공적인 결과를 거두고 있다.
- 이 서베이 논문은 통찰력있게 접근할 것이다. semantic , instance-level segmentation, fully convolutional pixel-labeling netowrks, encoder- decoder architectures, multi-scale and pyramid based approaches, recurrent networks, visual attention models, and gan 등등
- 유사성, 차이점 , 강점 등등 적용되고 있는 데이터셋 performances등에 대해서 논의할 예정이다
- 독자들에게 유망한 미래 연구 , 공부 방향도 제시해 줄 것이

# Introduction

### Image segmentation

1. Image segmentation의 중요성
- 장면을 이해하거나 , 비전으로 무언가를 이해할 때 필수적이다.
- Image(or video frames)을 여러개로 분활하는 것도 포함한다.
- 의료영상의 boundary extraction이나 measurement of tissue volumes autonomous vehicles 등등

1. Image segmentation algorithm :

% 전통적인 방법

- thresholding , histogram-based bundling, region-growing, k-means clustering, watersheds

% 비교적 최신

- active contours, graph cuts, conditional and Markov random fields, sparsity-based method
- Image segmentation 방법
    
    **thresholding**
    
    [3] N. Otsu, “A threshold selection method from gray-level histograms,” 
    
    **histogram-based bunding, region-growing**
    
    [4] R. Nock and F. Nielsen, “Statistical region merging,” 
    **K-means clustering**
    
    [5] N. Dhanachandra, K. Manglem, and Y. J. Chanu, “Image segmentation using k-means clustering algorithm and subtractive clustering algorithm,” 
    **watersheds**
    
    [6] L. Najman and M. Schmitt, “Watershed of a continuous function,”
    **to more advanced algorithms such as active contours**
    
    [7] M. Kass, A. Witkin, and D. Terzopoulos, “Snakes: Active contour models,” 
    **graph cuts**
    
    [8] Y. Boykov, O. Veksler, and R. Zabih, “Fast approximate energy minimization via graph cuts,” 
    **conditional and Markov random fields**
    
    [9] N. Plath, M. Toussaint, and S. Nakajima, “Multi-class image segmentation using conditional random fields and global classification,” 
    **sparsity-based**
    
    [10] J.-L. Starck, M. Elad, and D. L. Donoho, “Image decomposition via the combination of sparse representations and a variational approach,” 
    [11] S. Minaee and Y. Wang, “An admm approach to masked signal decomposition using subspace representation,” 
    **DeepLabv3**
    
    [12] L.-C. Chen, G. Papandreou, F. Schroff, and H. Adam, “Rethinking atrous convolution for semantic image segmentation,”
     
    

 ++ 최근 경향 : Deep learning 을 활용

1. Image segmentation의 종류

이미지 분할은 의미 레이블이 있는 픽셀의 분류 문제(시맨틱 분할) 또는 개별 객체의 분할(인스턴스 분할)이 있다. 의미론적 분할은 모든 이미지 픽셀에 대해 일련의 객체 범주(예: 사람, 자동차, 나무, 하늘)를 사용하여 픽셀 수준 레이블 지정을 수행하므로 일반적으로 전체 이미지에 대해 단일 레이블을 예측하는 이미지 분류보다 더 어려운 작업입니다. 인스턴스 분할은 이미지에서 관심 있는 각 객체를 감지하고 묘사함으로써 의미론적 분할 범위를 더욱 확장합니다.

> 우리는 최근 나온 이미지 segmentation 에 대해서 논의 할 것입니다 We provide a comprehensive review and insights on different aspects of these methods, including the training data, the choice of network architectures, loss functions, training strategies, and their key contributions. We present a comparative summary of the performance of the reviewed methods and discuss several challenges and potential future directions for
deep learning-based image segmentation models.
> 

### 분류기준 ( Contributions 기준으로 100개의 모델을 그룹화 함 )

1. Fully convolutional networks
2. Convolutional models with graphical models
3. Encoder-decoder based models
4. Multi-scale and pyramid network based models
5. R-CNN based models (for instance segmentation)
6. Dilated convolutional models and DeepLab family
7. Recurrent neural network based models
8. Attention-based models
9. Generative models and adversarial training
10. Convolutional models with active contour models
11. Other models

### 논문 요약

- 2019년까지 제안된 100개의 segmentation algorithms에 대하여 10 categories화 함
- 학습데이터, 아키텍처, loss function, 학습 전략, model 특성 등 많은 모델 review와 더불어 통찰력 있는 결과를 제시
- 20개의 유명한 데이터 셋에 대해 overview 할 것이다 (2D, 2.5D, 3D images)에 대해서

### 파트 요약

2) backbone 및 인기있는 architectures 요약

3) 100개 이상 모델들에 대한 개요 및 장점,특성 제공(segmentation)

4) 유명한 segmentation Dataset의 개요와 특성

5-1) Segmentation 성능 지표 검토

5-2) 모델들의 정량적 성능 검토( 성능 지표에 따라 )

6) 주요 과제와 해결방향 , 향후 

7) conclusion

# Overview of Deep Neural Networks

## 2.1 Convolutional Neural Networks

- CNN
    
    [13] Y. LeCun, L. Bottou, Y. Bengio, P. Haffner et al., “Gradient-based
    learning applied to document recognition,” Proceedings of the IEEE,
    vol. 86, no. 11, pp. 2278–2324, 1998.
    
    [17] K. Fukushima, “Neocognitron: A self-organizing neural network
    model for a mechanism of pattern recognition unaffected by shift
    in position,” Biological cybernetics, vol. 36, no. 4, pp. 193–202, 1980.
    [18] A. Waibel, T. Hanazawa, G. Hinton, K. Shikano, and K. J. Lang,
    “Phoneme recognition using time-delay neural networks,” IEEE
    transactions on acoustics, speech, and signal processing, vol. 37, no. 3,
    pp. 328–339, 1989.
    [19] A. Krizhevsky, I. Sutskever, and G. E. Hinton, “Imagenet classification with deep convolutional neural networks,” in Advances in
    neural information processing systems, 2012, pp. 1097–1105.
    [20] K. Simonyan and A. Zisserman, “Very deep convolutional
    networks for large-scale image recognition,” arXiv preprint
    arXiv:1409.1556, 2014.
    [21] K. He, X. Zhang, S. Ren, and J. Sun, “Deep residual learning
    for image recognition,” in Proceedings of the IEEE conference on
    computer vision and pattern recognition, 2016, pp. 770–778.
    [22] C. Szegedy, W. Liu, Y. Jia, P. Sermanet, S. Reed, D. Anguelov,
    D. Erhan, V. Vanhoucke, and A. Rabinovich, “Going deeper with
    convolutions,” in Proceedings of the IEEE conference on computer
    vision and pattern recognition, 2015, pp. 1–9.
    [23] A. G. Howard, M. Zhu, B. Chen, D. Kalenichenko, W. Wang,
    T. Weyand, M. Andreetto, and H. Adam, “Mobilenets: Efficient
    convolutional neural networks for mobile vision applications,”
    arXiv preprint arXiv:1704.04861, 2017.
    
    [24] G. Huang, Z. Liu, L. Van Der Maaten, and K. Q. Weinberger,
    “Densely connected convolutional networks,” in Proceedings of the
    IEEE conference on computer vision and pattern recognition, 2017, pp.
    4700–4708.
    

## 2.2 Recurrent Neural Networks

- RNN
    
    [14] S. Hochreiter and J. Schmidhuber, “Long short-term memory,”
    Neural computation, vol. 9, no. 8, pp. 1735–1780, 1997.
    [25] D. E. Rumelhart, G. E. Hinton, R. J. Williams et al., “Learning
    representations by back-propagating errors,” Cognitive modeling,
    vol. 5, no. 3, p. 1, 1988.
    

## 2.3 Encoder-Decoder and Auto-Encoder Models

[15] V. Badrinarayanan, A. Kendall, and R. Cipolla, “Segnet: A deep
convolutional encoder-decoder architecture for image segmentation,” IEEE transactions on pattern analysis and machine intelligence,
vol. 39, no. 12, pp. 2481–2495, 2017.

## 2.4 Generative Adversarial Networks(GANs)

[16] I. Goodfellow, J. Pouget-Abadie, M. Mirza, B. Xu, D. Warde-Farley,
S. Ozair, A. Courville, and Y. Bengio, “Generative adversarial
nets,” in Advances in neural information processing systems, 2014, pp.
2672–2680.

# DL-Based Image Segmentation

- 100 개의 segmentation method들을 리뷰 할거야! 10 카테고리 나눠서

### FCN

- 첫 번째 Deep learning works for semantic image segmentation, using fully convolutional network
    
    [31] J. Long, E. Shelhamer, and T. Darrell, “Fully convolutional networks for semantic segmentation,” 
    
- FCN은 conv layer 만을 포함하고 있다 image of arbitrary size를 받아서 Segmentation map of the same size :: 이미지 사이즈와 동일하게 segmentation map을 생성한다
- 원래 Google net 이나 VGG net 같은 경우에 출력층 부분에 FC layer를 활용하여 image classification을 진행하는데 Semantic segmentation 관점에서는 fc layer는 이미지의 위치정보가 사라지기 떄문에 사용에 한계가 있었다.
- 따라서 1. **완전 연결 계층을 구성하여 fc-layer를 Conv-Layer로 대체하여 Segmentation** 할 수 있도록 하였다
    
    1) Feature map 원본 이미지 위치 정보를 내포할 수 있다.
    
    2) Input size에 제한을 받지 않는다.
    
    ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/407c7a83-ee89-4431-bbde-16816d0a0fa3/Untitled.png)
    
- Skip connections in which feature maps from the final
    
    ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/7732efd0-4f99-476c-9bc5-0d8a8d66ba75/Untitled.png)
    
- 2.Transposed Convolution(흔히 말하는 Deconvolution)
    - Deconvolution은 convolutionㅇ르 거꾸로 수행하는 과정인데 Output image와 filter를 사용해 Input 이미지를 얻어내기 위한 역 수학 연산을 하는 과정 , 여기서는 Input image size와 동일한 size의 이미지를 얻기 위해 Convolution 과정을 그대로 수행하는 것 → 차이 “Transposed conv의 output은 Convolution의 input과 크기는 같지만 그 값은 다를 수 있다
- PASCAL VOC, NYUDv2, and SIFT Flow에 테스트를 했고
- FCN이 E2E에서 segmentation을 어떻게 접근할 수 있는지 쉬운 방법을 제안했지만 여전히 *느린 문제와 3D images로 transfer*하는 것에 어려움이 있었다.

### ParseNet

- [32] W. Liu, A. Rabinovich, and A. C. Berg, “Parsenet: Looking wider to see better,”
- FCN은 global context information을 무시 했는데 , ParseNet은 global context to FCNs 를 사용했다 average feature for a layer to augment the features at each loacation을 통해 !
    
    ![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/2aa1b00f-7519-4e61-989c-ccbfe421c77b/Untitled.png)
    

FCNs have been applied to a variety of segmentation
problems, such as brain tumor segmentation [33], instanceaware semantic segmentation [34], skin lesion segmentation [35], and iris segmentation [36].

- 33 ~ 36
    
    [33] G. Wang, W. Li, S. Ourselin, and T. Vercauteren, “Automatic brain
    tumor segmentation using cascaded anisotropic convolutional
    neural networks,” in International MICCAI Brainlesion Workshop.
    Springer, 2017, pp. 178–190.
    [34] Y. Li, H. Qi, J. Dai, X. Ji, and Y. Wei, “Fully convolutional instanceaware semantic segmentation,” in Proceedings of the IEEE Conference
    on Computer Vision and Pattern Recognition, 2017, pp. 2359–2367.
    [35] Y. Yuan, M. Chao, and Y.-C. Lo, “Automatic skin lesion segmentation using deep fully convolutional networks with jaccard
    distance,” IEEE transactions on medical imaging, vol. 36, no. 9, pp.
    1876–1886, 2017.
    [36] N. Liu, H. Li, M. Zhang, J. Liu, Z. Sun, and T. Tan, “Accurate
    iris segmentation in non-cooperative environments using fully
    convolutional networks,” in 2016 International Conference on
    Biometrics (ICB). IEEE, 2016, pp. 1–8.
    

## 3.6 Dilated Convolutional Models

# Image Segmentation Datasets

# Performance Review
